package swgemurpcserver.rpc;

import "InventoryItem.proto";

message SWGEmuCharacterDetail {
	required fixed64 object_id = 1;
	required fixed32 account_id = 2;
	required string first_name = 3;
	required string sur_name = 4;
	optional fixed32 bank_credits = 5;
	optional fixed32 cash_credits = 6;
	optional fixed32 base_health = 7;
	optional fixed32 base_action = 8;
	optional fixed32 base_mind = 10;
	optional string biography = 11;
	optional int32 remaining_plots = 12;
	optional int32 maximumim_plots = 13;
	optional string title = 14;
	
	repeated CharacterInventoryItem inventory_items = 20;
	repeated SWGEmuCharacterStructureItem structures = 21;
}

message SWGEmuCharacterStructureItem {
	required fixed64 object_id = 1;
	required string object_name = 2;
	required string display_name = 3;
}


message GetCharacterDetailsRequest {
	required fixed32 account_id = 1;
	optional fixed64 object_id = 2;
	optional fixed32 galaxy_id = 3;
	optional string first_name = 7;
	optional string sur_name = 8;
	optional string galaxy_name = 9;
}

message GetCharacterDetailsResponse {
	repeated SWGEmuCharacterDetail character_details = 1;
}

option cc_generic_services = true;

service SWGEmuCharacterDetailsService {
	rpc GetCharacterDetails (GetCharacterDetailsRequest) returns (GetCharacterDetailsResponse);
}